### DISCUSSION TITLE
Automerge not waiting for conditional github workflow

### DISCUSSION THREAD

**Discussion by ingkagero** [renovatebot/renovate#26413] [Request Help]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### What is your question?

Renovate automerge seems to ignore / not wait for a conditional job that publishes a docker image on certain code changes. I've set up renovate to automerge on patch updates and not set the ignoreTests parameter. I can't provide a log file at the moment as the last time it happened was 3 days ago and is no longer in the recent runs. The Github job that seems to get ignored is defined like this:

  publish:
    name: Publish
    needs: get-changed-services
    if: always() &&
      needs.get-changed-services.outputs.is_empty == 'false'
    uses: ./.github/workflows/publish.yml
    with:
      services: ${{needs.get-changed-services.outputs.matrix}}

Github's 'require checks to pass' doesn't work with the conditional job either, so I am using the poseidon/wait-for-status-checks action to enforce all checks for manual merges / github auto-merge.

Is there something I'm missing or a way to get renovate to also adhere to the enforce-all-checks action?

### Logs (if relevant)

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

**Comment by rarkins** [comment]
Are you using GitHub's platform-native automerge (default)? i.e. it's configured for GitHub automerge at creation.

When platform automerge is enabled, the platform decides when to merge and it's nothing to do with Renovate. Most likely you're using this and there's nothing Renovate can do differently to help you.

When Renovate does its own automerge then it will not automerge if there are any pending or failed checks.

**Reply by ingkagero** [reply]
Thank you for replying! Sorry I feel I described the original issue a bit poorly. The GH auto-merge can be enabled on an individual PR if GH the GH Setting under 'Pull Requests' allows for it. I'm not aware that there is an option to have this as a default - it is definitely not default in our case, just an option provided to the user. The GH auto-merge is also respecting the branch protection rules of the target, so no issues there.

The issue for me with PRs auto-merged by Renovate is, that those are consistently the only PRs that are seemingly bypassing branch-protections, i.e. are merged before the one check defined in branch protection is successful.

**Reply by rarkins** [reply]
We'll need logs to understand more then

**Reply by ingkagero** [reply]
Thanks you for looking at it! I will close this and open with a better description once I have some logs.
